infra/recipes/docker-compose/oh-hadoop-spark/jobs.yaml

```diff
@@ -11,60 +11,75 @@ jobs:
         "spark.sql.autoBroadcastJoinThreshold": "-1",
         "spark.driver.memory": "1g"
       }
-      jar-path: local:/opt/spark/openhouse-spark-apps_2.12-latest-all.jar
-      dependencies:
-        - local:/opt/spark/openhouse-spark-runtime_2.12-latest-all.jar
   spark:
-    engine-uri: http://spark-livy:8998
+    engines:
+      - &livy-engine
+        engine-type: LIVY
+        coordinator-class-name: com.linkedin.openhouse.jobs.services.livy.LivyJobsCoordinator
+        engine-uri: http://spark-livy:8998
+        jar-path: local:/opt/spark/openhouse-spark-apps_2.12-latest-all.jar
+        dependencies:
+          - local:/opt/spark/openhouse-spark-runtime_2.12-latest-all.jar
+    default-engine: LIVY
     storage-uri: http://openhouse-housetables:8080
     metrics-uri: http://localhost:4318
-    coordinator-class-name: com.linkedin.openhouse.jobs.services.livy.LivyJobsCoordinator
     auth-token-path: "/var/config/openhouse.token"
     apps:
       - type: NO_OP
         class-name: com.linkedin.openhouse.jobs.spark.NoOpSparkApp
         args: []
         << : *apps-defaults
+        << : *livy-engine
       - type: SQL_TEST
         class-name: com.linkedin.openhouse.jobs.spark.OpenHouseCatalogSQLTestSparkApp
         args: []
         << : *apps-defaults
+        << : *livy-engine
       - type: RETENTION
         class-name: com.linkedin.openhouse.jobs.spark.RetentionSparkApp
         args: []
         << : *apps-defaults
+        << : *livy-engine
       - type: DATA_COMPACTION
         class-name: com.linkedin.openhouse.jobs.spark.DataCompactionSparkApp
         args: []
         << : *apps-defaults
+        << : *livy-engine
```

Comment:
Do we need to pass default engine now as all the job type argument? Can we extract from the the engines tag?

Reply:
How about it this is not passed for all the job type can we automatically inter default engine?

Reply:
This is not passing the default engine, but passing the chosen engine. For example, it will be setu-engine and gg-engine. Default engine is used when this is not passed.